### PR TITLE
Switch to new host apis

### PR DESCRIPTION
**Pull request by HaoK**
Still working through the files

### CODE REVIEW

**Comment by HaoK** [comment]
I think I got them all, there were a lot of tests :/

File: src/Identity/test/InMemory.Test/FunctionalTest.cs [hunk 1]
```diff
@@ -285,10 +286,11 @@ private static async Task<XElement> GetAuthData(TestServer server, string url, s
...
         }
 
-        private static TestServer CreateServer(Action<IServiceCollection> configureServices = null, Func<HttpContext, Task> testpath = null, Uri baseAddress = null, bool testCore = false)
+        private static async Task<TestServer> CreateServer(Action<IServiceCollection> configureServices = null, Func<HttpContext, Task> testpath = null, Uri baseAddress = null, bool testCore = false)
         {
-            var builder = new WebHostBuilder()
-                .Configure(app =>
+            var host = new HostBuilder()
+                .ConfigureWebHost(builder =>
+                builder.Configure(app =>
```
Comment (BrennanConroy) [inline src/Identity/test/InMemory.Test/FunctionalTest.cs:293]: nit: formatting

File: src/Security/Authentication/test/CookieTests.cs [hunk 2]
```diff
@@ -214,7 +225,7 @@ public async Task SignOutWrongAuthTypeThrows()
         [Fact]
         public async Task CookieOptionsAlterSetCookieHeader()
         {
-            var server1 = CreateServer(o =>
+            var host = await CreateHost(o =>
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/CookieTests.cs:228]: nit: using

File: src/Security/Authentication/test/CookieTests.cs [hunk 3]
```diff
@@ -235,14 +247,15 @@ public async Task CookieOptionsAlterSetCookieHeader()
             Assert.Contains(" samesite=none", setCookie1);
             Assert.Contains(" httponly", setCookie1);
 
-            var server2 = CreateServer(o =>
+            var host2 = await CreateHost(o =>
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/CookieTests.cs:250]: nit: using

File: src/Security/Authentication/test/CookieTests.cs [hunk 4]
```diff
@@ -1000,19 +1044,24 @@ public async Task ChallengeDoesNotSet401OnUnauthorized()
...
-                    app.Run(context => context.SignInAsync(CookieAuthenticationDefaults.AuthenticationScheme, new ClaimsPrincipal(new ClaimsIdentity("whatever"))));
-                })
-                .ConfigureServices(services =>
-                {
-                    services.AddAuthentication().AddCookie();
-                    services.Configure<CookieAuthenticationOptions>(CookieAuthenticationDefaults.AuthenticationScheme,
-                        o => o.Cookie.Name = "One");
-                });
-            var server = new TestServer(builder);
+            var host = new HostBuilder()
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/CookieTests.cs:1047]: nit: using

File: src/Security/Authentication/test/CookieTests.cs [hunk 5]
```diff
@@ -1023,19 +1072,24 @@ public async Task CanConfigureDefaultCookieInstance()
...
-                    app.Run(context => context.SignInAsync("Cookie1", new ClaimsPrincipal(new ClaimsIdentity("whatever"))));
-                })
-                .ConfigureServices(services =>
-                {
-                    services.AddAuthentication().AddCookie("Cookie1");
-                    services.Configure<CookieAuthenticationOptions>("Cookie1",
-                        o => o.Cookie.Name = "One");
-                });
-            var server = new TestServer(builder);
+            var host = new HostBuilder()
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/CookieTests.cs:1075]: nit: using

File: src/Security/Authentication/test/CookieTests.cs [hunk 6]
```diff
@@ -1046,15 +1100,20 @@ public async Task CanConfigureNamedCookieInstance()
...
-            var builder = new WebHostBuilder()
-                .Configure(app =>
-                {
-                    app.UseAuthentication();
-                    app.Map("/notlogin", signoutApp => signoutApp.Run(context => context.SignInAsync("Cookies",
-                        new ClaimsPrincipal(new ClaimsIdentity("whatever")))));
-                })
-                .ConfigureServices(services => services.AddAuthentication().AddCookie(o => o.LoginPath = new PathString("/login")));
-            var server = new TestServer(builder);
+            var host = new HostBuilder()
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/CookieTests.cs:1103]: nit: using

few more in this file, I'll stop commenting

File: src/Security/Authentication/test/OpenIdConnect/TestServerBuilder.cs [hunk 7]
```diff
@@ -62,59 +63,63 @@ public static TestServer CreateServer(Action<OpenIdConnectOptions> options)
...
-                });
+                            services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+                                .AddCookie()
+                                .AddOpenIdConnect(options);
+                        }))
+                .Build();
 
-            return new TestServer(builder);
+            host.Start();
+            return host.GetTestServer(); ;
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/OpenIdConnect/TestServerBuilder.cs:122]: nit: remove `;`

File: src/Security/Authentication/test/WsFederation/WsFederationTest.cs [hunk 8]
```diff
@@ -43,20 +44,25 @@ public async Task VerifySchemeDefaults()
...
-                    {
-                        sharedOptions.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
-                        sharedOptions.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
-                        sharedOptions.DefaultChallengeScheme = WsFederationDefaults.AuthenticationScheme;
-                    })
-                    .AddCookie()
-                    .AddWsFederation();
-                });
-            var server = new TestServer(builder);
+            var host = new HostBuilder()
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/WsFederation/WsFederationTest.cs:47]: nit: using

File: src/Security/Authentication/test/WsFederation/WsFederationTest.cs [hunk 9]
```diff
@@ -198,30 +204,35 @@ public async Task RemoteSignoutRequestTriggersSignout()
...
-                        options.BackchannelHttpHandler = new WaadMetadataDocumentHandler();
-                        options.EventsType = typeof(MyWsFedEvents);
-                    });
-                })
-                .Configure(app =>
-                {
-                    app.Run(context => context.ChallengeAsync());
-                });
-            var server = new TestServer(builder);
+            var host = new HostBuilder()
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/WsFederation/WsFederationTest.cs:207]: nit: using

**Review by BrennanConroy — APPROVED** [review]
Thanks for doing this!

**Review by Tratcher — APPROVED** [review]

**Pull request merged**